bin/faucet/src/store.rs

```diff
@@ -44,27 +48,38 @@ impl FaucetDataStore {
     pub fn update_faucet_state(&self, new_faucet_state: Account) {
         *self.faucet_account.lock().expect("Poisoned lock") = new_faucet_state;
     }
+
+    /// Updates the stored faucet account with the new one.
+    pub fn load_transaction_script(&self, tx_script: &TransactionScript) {
+        // TODO: because the script string is interpolated, the MAST is different and needs to be
+        // loaded each time. Maybe it should be compiled once and inputs could be passed some other
+        // way
+        self.mast_store.insert(tx_script.mast().clone());
+    }
 }
 
 impl DataStore for FaucetDataStore {
     fn get_transaction_inputs(
         &self,
         account_id: AccountId,
-        _block_ref: BlockNumber,
-        _notes: &[NoteId],
-    ) -> Result<TransactionInputs, DataStoreError> {
+        _ref_blocks: BTreeSet<BlockNumber>,
```

Comment:
We don't care about reference blocks because we always execute against the genesis block?